via/static/scripts/video_player/hooks/use-app-layout.ts

```diff
@@ -0,0 +1,55 @@
+import type { RefObject } from 'preact';
+import { useCallback, useLayoutEffect, useState } from 'preact/hooks';
+
+/* Relative application container size */
+type AppSize = 'sm' | 'md' | 'lg' | 'xl' | '2xl';
+
+/**
+ * Return a string representing the relative size of `appContainer` that can
+ * be used to make decisions about app layout and sizing of components. This
+ * would typically be something handled by media queries, but it is an element
+ * (`body`) width that needs to be queried, not the viewport. We could replace
+ * this with container queries once browser support is sufficient.
+ */
+export function useAppLayout(appContainer: RefObject<HTMLDivElement | null>) {
+  const [appSize, setAppSize] = useState<AppSize>('sm');
+
+  function observeElementSize(
+    element: Element,
+    onSizeChanged: (width: number, height: number) => void
+  ): () => void {
+    const observer = new ResizeObserver(() =>
+      onSizeChanged(element.clientWidth, element.clientHeight)
+    );
+    observer.observe(element);
+    return () => observer.disconnect();
+  }
+
+  const updateWidth = useCallback(() => {
+    const containerWidth = appContainer.current?.clientWidth;
+    if (!containerWidth) {
+      return;
+    }
+
+    let containerSize: AppSize = 'sm';
+    if (containerWidth > 1376) {
+      containerSize = '2xl';
+    } else if (containerWidth > 1024) {
+      containerSize = 'xl';
+    } else if (containerWidth > 875) {
+      containerSize = 'lg';
+    } else if (containerWidth > 768) {
+      containerSize = 'md';
+    }
+
+    setAppSize(containerSize);
+  }, [appContainer]);
+
+  useLayoutEffect(() => {
+    const cleanup = observeElementSize(appContainer.current!, updateWidth);
+    updateWidth();
+    return cleanup;
+  }, [updateWidth, appContainer]);
```

Comment:
This layout effect will never re-run if the `appContainer.current` value changes, because the hook is watching the ref object rather than the `current` property.

Some options:

- Accept this limitation, but call it out in comments. I think this is the easiest option.
- Re-work hook to take an observable ref of some sort and then do the work to handle changes in the value internally. This effort is not justified at present IMO, unless we can find a really easy way of doing it.